src/components/shared/Button/ActionButton.tsx

```diff
@@ -4,32 +4,40 @@ import { MessageDescriptor } from 'react-intl';
 import { ActionTransformFnType } from '~utils/actions';
 import { useAsyncFunction, useMounted } from '~hooks';
 import DefaultButton from '~shared/Button';
+import { ActionTypes } from '~redux';
 import { Props as DefaultButtonProps } from './Button';
 
+const getFormAction = (
+  action: ActionTypes,
+  actionType: 'SUBMIT' | 'ERROR' | 'SUCCESS',
+) => {
+  const actionEnd = actionType === 'SUBMIT' ? '' : `_${actionType}`;
+  return `${action}${actionEnd}`;
+};
+
 interface Props extends DefaultButtonProps {
+  actionType: ActionTypes;
   button?: ElementType;
   confirmText?: any;
-  error: string;
   onConfirmToggled?: (...args: any[]) => void;
   onSuccess?: (result: any) => void;
-  submit: string;
-  success: string;
   text?: MessageDescriptor | string;
   transform?: ActionTransformFnType;
   values?: any | (() => any | Promise<any>);
 }
 
 const ActionButton = ({
+  actionType,
   button,
-  error,
-  submit,
-  success,
   onSuccess,
   // @todo Remove `values` once async transform functions are supported
   values,
   transform,
   ...props
 }: Props) => {
+  const submit = getFormAction(actionType, 'SUBMIT');
```

Comment:
These make more sense to live in here, right? Instead of duplicating this logic every time the ActionButton is used.